src/backend/distributed/sql/udfs/citus_check_connection_to_node/11.0-1.sql

```diff
@@ -0,0 +1,11 @@
+CREATE FUNCTION pg_catalog.citus_check_connection_to_node (
+    nodename text,
+    nodeport integer DEFAULT 5432)
+    RETURNS bool
+    LANGUAGE C
+    STRICT
```

Comment:
As we use `STRICT` if we provide NULL to nodename parameter, UDF will return NULL instantly. Is this the expected experience or should we remove the `STRICT` and add argument check for more explicit error message on UDF's definition?

Reply:
This is a valid argument. I guess it is better to not have NULL return values in any scenario.

We were planning to use `bool_and` aggregate on all the calls to `citus_check_connection_to_node` in the second phase of this task. To be able to do that, we should return false success values I guess.

The main problem is boolean logical operators tend to ignore NULL values.

```
postgres=# SELECT bool_and(x) FROM unnest(ARRAY[NULL, true]) x;
 bool_and
----------
 t
(1 row)

postgres=# SELECT bool_and(x) FROM unnest(ARRAY[NULL, false]) x;
 bool_and
----------
 f
(1 row) 

postgres=# SELECT bool_or(x) FROM unnest(ARRAY[NULL, false]) x;
 bool_or
---------
 f
(1 row)

postgres=# SELECT bool_or(x) FROM unnest(ARRAY[NULL, true]) x;
 bool_or
---------
 t
(1 row)
```

Reply:
I find the use of `STRICT` as a relief from code development purposes. If you pass `NULL` argument to a `STRICT` function, you should expect to get `NULL` values.

And, for our purposes, we'll use this on `pg_dist_node`, which is protected by `NOT NULL`
```SQL
 nodename         | text     |           | not null | 
 nodeport         | integer  |           | not null | 5432
```

So, I'd rather keep the STRICT for simplicity of the code.